src/runaqa.ts

```diff
@@ -1,324 +1,388 @@
+  }
+  tempDirectory = path.join(baseLocation, 'actions', 'temp')
+}
+
+export async function runaqaTest(
+  version: string,
+  jdksource: string,
+  customizedSdkUrl: string,
+  sdkdir: string,
+  buildList: string,
+  target: string,
+  customTarget: string,
+  aqatestsRepo: string,
+  openj9Repo: string,
+  tkgRepo: string,
+  vendorTestParams: string,
+  aqasystemtestsRepo: string
+): Promise<void> {
+
+  await setupTestEnv(
+    version,
+    jdksource,
+    customizedSdkUrl,
+    sdkdir,
+    buildList,
+    aqatestsRepo,
+    openj9Repo,
+    tkgRepo,
+    vendorTestParams,
+    aqasystemtestsRepo
+    );
+
+  const options: ExecOptions = {}
+  let myOutput = ''
+  options.listeners = {
+    stdout: (data: Buffer) => {
+      myOutput += data.toString()
+    }
+  }
+  process.chdir('TKG')
+  try {
+    await exec.exec('make compile')
+
+    if (target.includes('custom') && customTarget !== '') {
+      const customOption = `${target
+        .substr(1)
+        .toUpperCase()}_TARGET=${customTarget}`
+      await exec.exec('make', [`${target}`, `${customOption}`], options)
+    }
+    else if (target.includes('-f parallelList.mk')){
+      // move the parallelList to TKG/
+      if (IS_WINDOWS){
+        await io.cp(
+          `${process.env.GITHUB_WORKSPACE}\\parallelList.mk`,
+          `${process.env.GITHUB_WORKSPACE}\\aqa-tests\\TKG\\parallelList.mk`
+        )
+      } else {
+        await io.cp(
+          `${process.env.GITHUB_WORKSPACE}/parallelList.mk`,
+          `${process.env.GITHUB_WORKSPACE}/aqa-tests/TKG/parallelList.mk`
+        )
+      }
+      // Run the test
+      await exec.exec(`make ${target}`);
+   } else {
+      await exec.exec('make', [`${target}`], options)
+    }
+  } catch (error) {
+    if (error instanceof Error){
+      core.setFailed(error.message)
+    }
+    else{
+      core.setFailed('Unexpected error')
+    }
+  }
+  if (myOutput.includes('FAILED test targets') === true) {
+    core.setFailed('There are failed tests')
+  }
+}
+
+function resetJDKHomeFromProperties(): void {
+  const pfile = `${process.env.GITHUB_WORKSPACE}/aqa-tests/job.properties`
+  if (fs.existsSync(pfile)) {
+    const lines = fs
+      .readFileSync(pfile, 'utf-8')
+      .replace(/\r\n/g, '\n')
+      .split('\n')
+      .filter(Boolean)
+    for (const l of lines) {
+      const regexp = /TEST_JDK_HOME=(.*)/
+      const match = regexp.exec(l)
+      if (match && match[1]) {
+        process.env.TEST_JDK_HOME = match[1]
+        core.info(`Reset TEST_JDK_HOME to ${process.env.TEST_JDK_HOME}`)
+      }
+    }
+  }
+}
+
+function getTestJdkHome(version: string, jdksource: string): string {
+  // Try JAVA_HOME first and then fall back to GITHUB actions default location
+  let javaHome = process.env[`JAVA_HOME_${version}_X64`] as string
+  if (javaHome === undefined) {
+    javaHome = process.env['JAVA_HOME'] as string
+  }
+  if (jdksource === 'install-jdk') {
+    // work with AdoptOpenJDK/install-sdk

```

Comment:
Is that still AdoptOpenJDK or Adoptium?

Reply:
Will update.

Reply:
I believe this comment means that we are using the AdoptOpenJDK/install-jdk action. So I do not think there is anything to update here.

https://github.com/AdoptOpenJDK/install-jdk --> Action is still under AdoptOpenJDK/install-jdk

Reply:
correct, install-jdk did not get transitioned over to Eclipse Foundation... also, at some point we may stop using it (as per https://github.com/adoptium/run-aqa/issues/122), but for now please leave as is